### PR TITLE
Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by shoracek**
Addresses issue #114.

### CODE REVIEW

**Comment by tohojo** [comment]
Awesome, thanks! Do you know if there is some repository configuration that needs to happen to enable this, or is it enough to just merge the PR?

File: xdp-dump/tests/test-xdpdump.sh [hunk 1]
```diff
@@ -341,7 +341,7 @@ test_multi_pkt()
     for PKT_SIZE in "${PKT_SIZES[@]}" ; do
 
         PID=$(start_background_no_stderr "$XDPDUMP -i $NS -p xdp_test_prog_with_a_long_name --rx-capture=entry,exit")
-        timeout 4 $PING6 -W 2 -s "$PKT_SIZE" -c 20000 -f "$INSIDE_IP6" || return 1
+        $PING6 -W 2 -s "$PKT_SIZE" -c 20000 -f "$INSIDE_IP6" || return 1
```
Comment (chaudron) [inline xdp-dump/tests/test-xdpdump.sh:344]: The timeout should not be deleted, as it could cause the test to run for an extremely long time on failure.

Comment (tohojo) [reply xdp-dump/tests/test-xdpdump.sh:344]: We could increase it to something that should never be hit in practice but still won't leave the test running forever? Like 30 seconds or so?

Comment (chaudron) [reply xdp-dump/tests/test-xdpdump.sh:344]: Guess the question was, why was it removed? Was it causing a timeout before the test was actually completed?

If so, yes then changing it to "timeout 30 ..." would be better than not having it.

Comment (shoracek) [reply xdp-dump/tests/test-xdpdump.sh:344]: While initially testing it, even after increasing the time, it sometimes caused a timeout. But now, I cannot reproduce that anymore, so I will just use the longer timeouts.

**Comment by tohojo** [comment]
Ah, awesome, and now the tests also ran on the PR; looks good! Thanks a bunch for fixing this! :)

**Comment by tohojo** [comment]
Right, so the first test-run after merging failed...
https://github.com/xdp-project/xdp-tools/runs/2904042782?check_suite_focus=true

@chaudron any thoughts on how to make that test more robust?

**Comment by chaudron** [comment]
This is not the test with the timeout. Not sure why it's not receiving the packet. Maybe it takes longer for xdpdump to load, and hence the packets arrive before it's ready? Maybe the sleep 1, needs to be sleep 2 in start_background()?

**Comment by tohojo** [comment]
Eelco Chaudron ***@***.***> writes:

> This is not the test with the timeout. Not sure why it's not receiving
> the packet. Maybe it takes longer for xdpdump to load, and hence the
> packets arrive before it's ready? Maybe the sleep 1, needs to be sleep
> 2 in start_background()?

Hmm, would be great if we could rely less on random sleeps and more on
proper serialisation; but that would require xdpdump to signal readiness
somehow, I suppose?

**Comment by shoracek** [comment]
Doesn't xdpdump print a string after it starts listening? Maybe just waiting until the output file contains that string could be enough.

**Comment by chaudron** [comment]
There is the "listening on...." message which is printed when done setting up all stuff, but it might be buffered by the redirected output. We could use bpftool to see if the tracepoint is attached. All this assuming this is causing the failure ;)

**Comment by tohojo** [comment]
Seems just increasing the sleep time at least fixed it for now :)

**Pull request merged**